### PR TITLE
Include lambda guidelines

### PR DESCRIPTION
**Pull request by ball-hayden**
Rails 4 will see a large increase in the number of lambdas used, as scopes with a hash are deprecated.

The following is built from a brief chat in #engineering, although further comment would be welcomed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -342,6 +342,30 @@ While several Ruby books suggest the first style, the second is much more promin
...
+- For multi-line lambdas, prefer `lambda do ... end`
+
+```ruby
+  # bad
+  scope :long_example, lambda {
+    # ...
+  }
+
+  # good
+  scope :long_example, (lambda do
```
Comment (davepilling) [inline README.md]: Are the parentheses needed?

Comment (ball-hayden) [inline README.md]: Unfortunately, I think they are.

I've suggested preferring `do ... end` as that's what [The Style Guide](https://github.com/bbatsov/ruby-style-guide) recommends, although I do find this a little messy when the lambda is an argument.

Comment (jaybee) [inline README.md]: Unless there's a strong argument against this, I'd say we stick to `{ }` for all `lambda` declarations. I'm actually not sure I've ever seen `lambda do … end` in the wild

Comment (jaybee) [inline README.md]: Actually, looking at the upstream style guide, there's no explicit "use `do … end`" rule — it's just implied by the examples (there's no "bad" version). I'd say we stay equally vague here and assume people will be able to use their judgement when dealing with multi-line `lambdas`. The key parts really are about when and how to prefer stabby lambdas

Comment (caius) [inline README.md]: http://devblog.avdi.org/2011/07/26/the-procedurefunction-block-convention-in-ruby/ might be an interesting read to distill into our style guide. It's mostly (mostly!) the way I write blocks. (Although I simplify it usually to `do/end` for multiline, unless I'm chaining or `{}` stops me using parens.)

**Pull request closed without merging**